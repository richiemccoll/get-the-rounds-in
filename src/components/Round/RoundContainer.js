import React, { Component } from "react";
import { Confirm } from "semantic-ui-react";
import PropTypes from "prop-types";
import Round from "./Round";

class RoundContainer extends Component {
  constructor(props) {
    super(props);

    this.state = {
      open: false,
      result: ""
    };

    this.removeDrink = this.removeDrink.bind(this);
    this.show = this.show.bind(this);
    this.handleConfirm = this.handleConfirm.bind(this);
    this.handleCancel = this.handleCancel.bind(this);
    this.getTime = this.getTime.bind(this);
  }

  removeDrink(item) {
    this.props.removeDrink(item);
  }

  getTime() {
    const time = new Date().getTime();
    const date = new Date(time);
    return date;
  }

  show() {
    this.setState({ open: true });
  }

  handleConfirm(total) {
    const order = {
      id: Math.random().toString(36).slice(2),
      bar: this.props.bar,
      time: this.getTime(),
      drinks: this.props.round,
      total: total
    };

    this.props.confirmRound(order);
    this.props.clearRound();
    this.props.showNotification("Your order has been placed");

    this.setState({ result: "confirmed", open: false });
  }

  handleCancel() {
    this.setState({ result: "cancelled", open: false });
  }

  render() {
    const prices = this.props.round.map(i => i.price);
    const total = prices.reduce((total, acc) => total + acc);

    return (
      <div>
        <Round
          total={total}
          round={this.props.round}
          removeDrink={this.removeDrink}
          confirm={this.show}
        />
        <Confirm
          open={this.state.open}
          onCancel={this.handleCancel}
          onConfirm={this.handleConfirm}
        />
      </div>
    );
  }
}

export default RoundContainer;

RoundContainer.PropTypes = {
  round: PropTypes.array.isRequired,
  removeDrink: PropTypes.func.isRequired
};
