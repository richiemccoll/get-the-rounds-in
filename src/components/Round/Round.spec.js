import React from "react";
import RoundContainer from "./RoundContainer";
import renderer from "react-test-renderer";

describe("Round Component", () => {
  const selectedBar = {
    id: 1,
    image_url: "./potstill.jpg",
    name: "The Pot Still",
    address: "154 Hope Street",
    description: "A real local favourite, receiving more shout-outs from our followers than anywhere else. ",
    lat: 0,
    long: 0,
    text: "The Pot Still",
    value: "0"
  };

  const currentRound = [{ drink: 'Irn Bru', price: 2.7 }];

  const removeDrink = () => console.log("removing drink");
  const confirmRound = () => console.log("confirming round");
  const clearRound = () => console.log("clearing round");
  const showNotification = () => console.log("showing notification");

  it("should render the current round items", () => {
    const component = renderer.create(
      <RoundContainer
        bar={selectedBar.text}
        round={currentRound}
        removeDrink={removeDrink}
        confirmRound={confirmRound}
        clearRound={clearRound}
        showNotification={showNotification}
      />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
