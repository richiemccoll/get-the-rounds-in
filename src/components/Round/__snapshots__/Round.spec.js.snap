// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Round Component should render the current round items 1`] = `
<div>
  <div>
    <div
      className="ui divider"
    />
    <h3
      className="ui header"
    >
      Current Round Total: Â£
      2.70
    </h3>
    <h4
      className="ui header"
    >
      Irn Bru
      <i
        className="remove circle icon"
        onClick={[Function]}
      />
    </h4>
    <button
      className="ui button"
      onClick={[Function]}
      tabIndex={undefined}
    >
      Confirm Round
    </button>
    <div
      className="ui divider"
    />
  </div>
</div>
`;
